concepts/objects/about.md

```diff
@@ -0,0 +1,251 @@
+# About
+
+## Explanation
+
+Besides primitive data types like `number` and `string`, there is another important data type in JavaScript called `object`.
+Objects are collections of key-value pairs.
+In other languages they are referred to as maps or dictionaries and often times values need to have the same data type.
+In JavaScript, only the type of the key is restricted, it has to be a string.
+The values inside one object can have different types.
+They can be primitive types like numbers but also arrays, other objects or even functions.
+
+## Creating an Object
+
+You create an object (literal) with curly brackets.
+You can also directly include some entries (key-value pairs).
+For that, state the key first, followed by a colon and the value.
+
+```javascript
+const emptyObject = {};
+
+const obj = {
+  favoriteNumber: 42,
+  greeting: 'Hello World',
+  useGreeting: true,
+  address: {
+    street: 'Trincomalee Highway',
+    city: 'Batticaloa',
+  },
+  fruits: ['melon', 'papaya'],
+  addNumbers: function (a, b) {
+    return a + b;
+  },
+};
+```
+
+The trailing comma after the last entry is optional in JavaScript.
+
+You might wonder why the keys are not wrapped in quotation marks although they are supposed to be strings.
+This is a short-hand notation.
+If the key follows the naming rules for a JavaScript [identifier][mdn-identifier], you can omit the quotation marks.
+For keys with special characters in the name, you need to apply the usual string notation.
+
+```javascript
+const obj = {
+  '1keyStartsWithNumber': '...',
+  'key/with/slashes': '...',
+  'key-with-dashes': '...',
+  'key with spaces': '...',
+  '#&()[]{}èä樹keyWithSpecialChars': '...',
+};
+```
+
+Often times you want to combine existing variables into an object.
+This would often lead to key-value pairs like `name: name`.
+JavaScript has a short-hand notation that allows to just write `name` instead.
+There is also a short-hand notation for defining a function in an object.
+There you can omit the colon and the `function` keyword.
+
+```javascript
+const x = 1;
+const y = 2;
+
+const obj = {
+  x: x,
+  y: y,
+  calcSum: function (a, b) {
+    return a + b;
+  },
+};
+
+// Can be shortened to ...
+const obj = {
+  x,
+  y,
+  calcSum(a, b) {
+    return a + b;
+  },
+};
+```
+
+If you want to create an object literal with a key name that is stored in a variable or derived via an expression, you need to wrap the variable or expression in square brackets.
+
+```javascript
+const key = 'greeting';
+
+const obj = {
+  [key]: 'hello world',
+};
+```
+
+## Retrieving a Value
+
+There are two ways to retrieve the value for a given key, dot notation and bracket notation.
```

Comment:
We should definitely NOT mention this, but there are more than two ways.

```javascript
const obj = { greeting: 'hello world' }

const descriptor = Object.getOwnPropertyDescriptor(obj , 'greeting')
// => {value: "hello world", writable: true, enumerable: true, configurable: true}

descriptor.value
// => "hello world"
```

And naturally by way of destructuring.

There's even `Object.values`:

```javascript
const obj = { foo: 'bar', greeting: 'hello world' }

const values = Object.values(obj)
const index = Object.getOwnPropertyNames(obj).indexOf('greeting')

values[index]
// => "hello world"
```

By placing these in this comment on this PR, someone, someday might find them and be sad or happy.

Reply:
"There are some more ways ... " probably applies to every other thing I cover in all the JavaScript concepts. :wink: